### ISSUE TITLE
sorted() does not enforce comparable items in an iterable of tuples

### ISSUE THREAD

**Issue by jcmacdon** [python/typeshed#13297]
`sorted()` does not enforce that every item in the tuple must be comparable:
```python
items: list[tuple[int, object]] = [(1, object()), (1, object())]
sorted(items)
```

https://github.com/python/typeshed/blob/e5c531811536a85c8838ef0862eeb19256ecaf90/stdlib/builtins.pyi#L1762-L1767

**Comment by ajaya0** [comment]
Hi,
Can you explain the issue in more detail so that I can resolve it?

**Comment by jcmacdon** [comment]
In my example `items: list[tuple[int, object]]` should not be sortable because it is a tuple which contains `object` which is not comparable.

I think the main issue here is actually not `sorted()` in particular but the fact that any tuple appears to be compatible with `SupportsRichComparison`.

This code typechecks ok:
```python
from _typeshed import SupportsRichComparison

tuple1: SupportsRichComparison = (1, object())
tuple2: SupportsRichComparison = (1, object())

items = [tuple1, tuple2]
sorted(items)
```
but if you run it, you will get a runtime error:
```python
TypeError: '<' not supported between instances of 'object' and 'object'
```

**Comment by ajaya0** [comment]
Hi @jcmacdon,

Thanks for the clarification! I see the issue now. The sorted() function was allowing tuples containing non-comparable types to be sorted, which caused the TypeError at runtime.

**Comment by Daverball** [comment]
This is a problem with the `SupportsRichComparison` protocol in general, it's difficult to frame comparability purely as a type, so I don't think it will be possible to get rid of this false negative without type inversion.

You'd need something like this recursive type definition with type intersection and type inversion to handle this case properly

```python
type Comparable = (SupportsRichComparison & Not[tuple[Any, ...]) | tuple[Comparable, ...]
```

**Comment by srittau** [comment]
I agree that type checkers should complain about the example and the example given by @KotlinIsland in #13499: 

```python
sorted(["a", 1]) 
```

But per @Daverball, I don't think there is anything that we can do to improve the situation without changes to type system. So I'm closing this as inexpressible for now. If you have any ideas that we have overlooked, please send an exploratory PR.
